### ISSUE TITLE
Add new data quality or data profiling UI enhancements

### ISSUE THREAD

**Issue by thadguidry** [OpenRefine/OpenRefine#5315]
Facets are great for quickly getting a sense of a column's values and showing the count of each unique value.
However, there are broad strokes of general data quality or data profiling views in other tools and the OpenRefine ecosystem of extensions and addons that could be looked to as inspiration to incorporate some basic and advanced data profiling/data quality visualizations in OpenRefine's core.

### Proposed solution
Enable new **interactive** overlay modeling or view(s) to provide data profiling and data quality checking.
Example screenshot from extension, OpenRefineQualityMetrics

![image](https://user-images.githubusercontent.com/986438/194249763-976fe11c-15c7-4713-998a-584074320422.png)

#### Data Profiling
With data profiling it would be statistical based to showcase error distribution heatmaps, and other visualizations later.  Example existing OpenRefine extension: https://github.com/christianbors/OpenRefineQualityMetrics
- this might take the form of existing usage of our Facet area in new ways.
- or perhaps better visually allow Facets or Analytic Views to lock into place above their respective columns.  The data grid view would essentially be split horizontally with an upper and lower.
- dockable or movable Facets/Views might be needed for some analytical visualizations for profiling different data types.
#### Data Quality
The data quality checking (rules) should support some initial common formats and (sub)structures that many users expect and use.
- the formats or data sub structures can be maintained as predefined in the core of OpenRefine and extensible via a secondary UI to write rules with a `name` and `expression`.
- users can pick the rules they want to do data quality checking against each column.
- allowing rules to be added or removed as well as enabled/disabled will be important to users.
- have a centralalized browsing experience to find new rules added by the community and filter by categories/domain to allow a user to add them into OpenRefine's Data Quality checking.
- Checks or Rules are likely long running operations sometimes and should be supported especially if the data quality checks involve eager expressions or regex lookarounds, especially negative lookbehind.

### Alternatives considered
Instead of core, make this a robust extension.  The extension might warrant support from diverse groups that support a OpenRefine grant(s) for the extension.  Or directly help and support @christianbors with his extension and expand it's features which might take the form of gifted support from OpenRefine's own grants in the future.

### Additional context
@magdmartin @thadguidry @tfmorris have some small previous history with this proposal and discussion archived in our Google Groups mailing list (somewhere!!)  If I or someone else finds it, link it here please.

**Comment by ostephens** [comment]
I really like the idea of having support for this type of data checking built into the core product - it's the type of thing I often use a custom text facet to achieve. Because of the variety of uses and users for OpenRefine, having the ability to define what checks are done on a particular column would be really important IMO. However I could see it being good to deliver some basic checks by default - perhaps specifically data type (String/Number/Date/Boolean) as this is something that seems to come up reasonably regularly from users.

I've not had a chance to experiment with the OpenRefineQualityMetrics extension yet but it looks quite amazing.

**Comment by thadguidry** [comment]
@ostephens yes, me too! on using custom text facet for same thing!  Regarding the variety of uses/users, I have a UI design in my head for finding the checks/rules... it would be tag friendly and allow users to filter data checks by multiple tags, similar to how our project view works. I've used this sort of system in other tools like Talend Open Studio, etc.  It also would do previewing so users quickly can see how the check works against their data values or not. This would look and work somewhat similar to RegExr and Talend maybe Trifacta and a little less like Informatica Powercenter and Aggregate Profiler.  Microsoft Azure Purview doesn't even have this yet! 

> Azure Purview does not currently have any data profiling capabilities. It is something we are exploring in our roadmap. No ETA at this time

**Data Profiling** has many forms but ultimately, it's about gathering statistics and visualizations around data structures.
Some forms are:
1. Statistics like min,max,count,sum
2. Frequency distribution by (% of natural keys/distinct values, Length, Types, Patterns, % of zero/blank/null )
3. Foreign Key / Private Key relationships / Orphan Keys
4. Cardinality - checking relations of 1-1, 1-N, N-N

**Data Quality** checks definitely are rules based on any user/org but ultimately are Pattern checks at the end of the day to flag rows/records and take more action.  The checks could be part of a step to perform in a workflow (although our workflows are currently limited - no STOP_IF_FAILS), it could align with @wetneb proposal in his paper for more reusable step based transformations and UI down the line.
